### PR TITLE
fix(VirtualScroll): initialise differ with trackByFn

### PR DESCRIPTION
**Pull request by masimplo**
#### Short description of what this resolves:
VirtualScroll trackByFn has no effect due to the random order that Angular interprets input values.

#### Changes proposed in this pull request:

Virtual TrackByFn is changed into as setter input, that redefines the change detection mechanism as soon as trackBy is parsed, to take into account the provided function

**Ionic Version**: 3.x

**Fixes**: #7531
Related to #11294

### CODE REVIEW

File: src/components/virtual-scroll/virtual-scroll.ts [hunk 1]
```diff
@@ -368,7 +369,13 @@ export class VirtualScroll implements DoCheck, AfterContentInit, OnDestroy {
   /**
    * @input {function} Same as `ngForTrackBy` which can be used on `ngFor`.
    */
-  @Input() virtualTrackBy: TrackByFn;
+  @Input() set virtualTrackBy(val: TrackByFn) {
+    if (!isPresent(val)) return;
+    this._virtualTrackBy = val;
+    if (isPresent(this._records)) {
+      this._differ = this._iterableDiffers.find(this._records).create(this.virtualTrackBy);
```
Comment (manucorporat) [inline src/components/virtual-scroll/virtual-scroll.ts]: can we move the this._differ creation logic to a new function? maybe `_updateDiffer`. it is duplicated in here and virtualScroll.

Also, can you add a new-line after @Input()?

Comment (masimplo) [inline src/components/virtual-scroll/virtual-scroll.ts]: Wasn't sure if you prefer I overwrite last commit or add a new one, so I went with the safer choice.

**Comment by plusCubed** [comment]
The second commit appears to break the fix - the `virtualScroll` input setter is called first, which calls `_updateDiffer` and initializes`_differ` with the default `virtualTrackBy` function. When the `virtualTrackBy` input setter is called later, `_differ` is no longer blank.

**Comment by masimplo** [comment]
Oh yes. You are correct. isBlank(this.differ) check should not be there.

**Pull request merged**